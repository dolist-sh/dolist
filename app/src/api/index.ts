//TODO: Add return data type
export const getUser = async (token: string) => {
  const response = await fetch(`http://localhost:8000/user`, {
    method: 'GET',
    mode: 'cors',
    headers: { Authorization: `token ${token}` },
  });
  return response.json();
};

export const getAuthCode = async (sessionCode: string): Promise<string | null> => {
  const response = await fetch(`http://localhost:8000/auth?session_code=${sessionCode}`, {
    method: 'POST',
    mode: 'cors',
  });

  if (response.status === 200) {
    return response.json();
  }

  return null;
};

export const getUserRepos = async (token: string) => {
  const response = await fetch(`http://localhost:8000/user/repos`, {
    method: 'GET',
    mode: 'cors',
    headers: { Authorization: `token ${token}` },
  });
  return response.json();
};
