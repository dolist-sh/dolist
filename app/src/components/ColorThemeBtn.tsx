import * as React from 'react';

interface ColorThemeBtnProps {
  currentTheme: string;
  colorThemeIconUri: string;
  clickHandler: (event: React.MouseEvent) => void;
}

const ColorThemeBtn: React.FC<ColorThemeBtnProps> = ({
  clickHandler,
  colorThemeIconUri,
  currentTheme,
}: ColorThemeBtnProps) => {
  return (
    <button
      onClick={clickHandler}
      className="inline-flex w-[120px] h-[35px] bg-dolist-cream dark:bg-dolist-darkblue text-sm border-dolist-darkblue dark:border-dolist-cream border-2 rounded-md justify-center items-center"
    >
      <p className="pr-2 text-xs text-dolist-darkblue dark:text-dolist-cream font-std font-bold">
        {currentTheme === 'dark' ? 'Light Mode' : 'Dark Mode'}
      </p>
      <img src={colorThemeIconUri} className="w-[12px] h-[12px]" />
    </button>
  );
};

export default ColorThemeBtn;
