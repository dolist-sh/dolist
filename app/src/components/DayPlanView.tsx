
/* This example requires Tailwind CSS v2.0+ */
import * as React from 'react';

const DayHeader: React.FC = () => {
    return (
        <header className="relative z-20 flex flex-none items-center justify-between border-b border-gray-200 py-4 px-6">
          <div>
            <h1 className="text-lg font-semibold leading-6 text-gray-900">
              <time dateTime="2022-01-22" className="sm:hidden">
                Jan 22, 2022
              </time>
              <time dateTime="2022-01-22" className="hidden sm:inline">
                January 22, 2022
              </time>
            </h1>
            <p className="mt-1 text-sm text-gray-500">Saturday</p>
          </div>
        </header>
    );
}

const DayGrid: React.FC = () => {
    return (
        <div className="inline-flex flex-auto h-screen overflow-auto bg-white">
          <div className="flex flex-auto flex-col">
            <div className="flex w-full flex-auto">
              <div className="w-14 flex-none bg-white ring-1 ring-gray-100" />
              <div className="grid flex-auto grid-cols-1 grid-rows-1">
                {/* Horizontal lines */}
                <div
                  className="col-start-1 col-end-2 row-start-1 grid divide-y divide-gray-100"
                  style={{ gridTemplateRows: 'repeat(48, minmax(3.5rem, 1fr))' }}
                >
                  <div className="row-end-1 h-7"></div>
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      12AM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      1AM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      2AM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      3AM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      4AM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      5AM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      6AM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      7AM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      8AM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      9AM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      10AM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      11AM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      12PM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      1PM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      2PM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      3PM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      4PM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      5PM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      6PM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      7PM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      8PM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      9PM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      10PM
                    </div>
                  </div>
                  <div />
                  <div>
                    <div className="sticky left-0 -mt-2.5 -ml-14 w-14 pr-2 text-right text-xs leading-5 text-gray-400">
                      11PM
                    </div>
                  </div>
                  <div />
                </div>
                {/* Horizontal lines */}

              </div>
            </div>
          </div>
        </div>
    )
}


const DayPlanView: React.FC = () => {      
    return (
      <div className="flex h-full flex-col">
        <DayHeader />
        <DayGrid />
      </div>
    )
    
}

export default DayPlanView;
