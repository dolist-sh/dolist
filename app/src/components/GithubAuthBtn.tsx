import * as React from 'react';

interface GithubAuthBtnProps {
  oauthUri: string;
  logoUri: string;
}

const GithubAuthBtn: React.FC<GithubAuthBtnProps> = ({ oauthUri, logoUri }: GithubAuthBtnProps) => {
  return (
    <button
      className="inline-flex w-[220px] h-[45px] px-3 py-2 bg-dolist-cream dark:bg-dolist-darkblue border text-sm border-black dark:border-dolist-cream rounded  justify-center items-center"
      type="button"
    >
      <img src={logoUri} alt="github_logo_black_color" className="w-[28px] h-[28px]" />
      {/** TODO: change this to p tag when click handler is implemented */}
      <a href={oauthUri} className="pl-4 text-xs text-black dark:text-white font-std font-bold">
        {'Continue with Github'}
      </a>
    </button>
  );
};

export default GithubAuthBtn;
