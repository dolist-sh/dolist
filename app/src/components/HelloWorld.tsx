import * as React from 'react';

const HelloWorld: React.FunctionComponent = () => {
  return (
    <div className="flex min-h-screen flex-col items-center justify-center py-2">
      <h1 className="text-3xl underline">Hello World!</h1>
    </div>
  ); 
};

export default HelloWorld;
