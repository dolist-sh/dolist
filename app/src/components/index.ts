export { default as TaskArea } from './TaskArea';
export { default as TaskCard } from './TaskCard';
export { default as RepoCard } from './RepoCard';
export { default as Loader } from './Loader';
export { default as GithubAuthBtn } from './GithubAuthBtn';
export { default as ColorThemeBtn } from './ColorThemeBtn';
export { default as Layout } from './Layout';
export { default as AddRepoModal } from './AddRepoModal';
export { default as MonitoredRepoOverview } from './MonitoredRepoOverview';
export { default as MonitoredRepoDetail } from './MonitoredRepoDetail';
