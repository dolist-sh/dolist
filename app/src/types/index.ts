export type Repo = {
  id: number;
  name: string;
  defaultBranch: string;
  language: string;
  url: string;
};
