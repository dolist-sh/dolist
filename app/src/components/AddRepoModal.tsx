/* This example requires Tailwind CSS v2.0+ */
import * as React from 'react';
import { Fragment, useState, useEffect } from 'react';
import { Transition } from '@headlessui/react';
import { XIcon } from '@heroicons/react/outline';
import { getGithubRepos, postMonitoredRepos } from '../api';
import RepoCard from './RepoCard';

interface AddRepoModalProps {
  openCounter: number;
  githubLogoUri: string;
}

const AddRepoModal: React.FC<AddRepoModalProps> = ({ openCounter, githubLogoUri }: AddRepoModalProps) => {
  const [open, setOpen] = useState(false);
  const [repos, setRepos] = useState(null);

  useEffect(() => {
    openCounter > 0 ? setOpen(!open) : null;
  }, [openCounter]);

  useEffect(() => {
    const token = localStorage.getItem('token') as string;

    if (open) {
      getGithubRepos(token).then((data) => {
        setRepos(data);
      });
    }
  }, [open]);

  // TODO: Implement the handler with API call
  const addRepoSubmitHandler = async (event: React.MouseEvent) => {
    event.preventDefault();
    const token = localStorage.getItem('token');
    await postMonitoredRepos(token);
  };

  return (
    <Transition.Root show={open} as={Fragment}>
      <div className="relative z-10">
        <Transition.Child
          as={Fragment}
          enter="ease-out duration-300"
          enterFrom="opacity-0"
          enterTo="opacity-100"
          leave="ease-in duration-200"
          leaveFrom="opacity-100"
          leaveTo="opacity-0"
        >
          <div onClick={() => setOpen(false)} className="fixed inset-0 bg-gray-500 bg-opacity-75 transition-opacity" />
        </Transition.Child>
        <div className="fixed flex left-[50%] translate-x-[-50%]">
          <Transition.Child
            as={Fragment}
            enter="ease-out duration-300"
            enterFrom="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
            enterTo="opacity-100 translate-y-0 sm:scale-100"
            leave="ease-in duration-200"
            leaveFrom="opacity-100 translate-y-0 sm:scale-100"
            leaveTo="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
          >
            <div className="relative bg-white dark:bg-dolist-bg-dark rounded px-4 pt-5 pb-4 text-left overflow-hidden shadow-xl transform transition-all sm:my-8 sm:max-w-lg sm:w-full sm:p-6">
              <div className="hidden sm:block absolute top-0 right-0 pt-4 pr-4">
                <button
                  type="button"
                  className="bg-white rounded-md text-black dark:text-white focus:outline-none"
                  onClick={() => setOpen(false)}
                >
                  <span className="sr-only">Close</span>
                  <XIcon className="h-6 w-6 bg-white dark:bg-dolist-bg-dark" aria-hidden="true" />
                </button>
              </div>
              <div className="sm:flex sm:flex-col sm:items-start">
                <div className="mt-3 text-center sm:mt-0 sm:ml-4 sm:text-left">
                  <h3 className="text-lg leading-6 font-std font-bold text-black dark:text-white">
                    {`Add Repositories`}
                  </h3>
                </div>
                <div className="flex flex-col w-[95%] ml-4 mt-7 border-b border-dolist-lightgray dark:border-dolist-cream">
                  <div className="flex flex-row pb-1">
                    <p className="cursor-pointer text-xs font-std font-bold text-dolist-gray dark:text-dolist-cream pr-3">{`whathecker`}</p>
                    <p className="cursor-pointer text-xs font-std hover:font-bold text-dolist-gray dark:text-dolist-cream pr-3">{`username_1`}</p>
                  </div>
                </div>
              </div>
              <div className="flex flex-col w-[95%] ml-4 mt-4 min-h-[200px] max-h-[250px] overflow-hidden">
                <div className="flex flex-col w-full overflow-y-scroll">
                  {/** TODO: Add Loader */}
                  {repos
                    ? repos.map((repo, index) => {
                        return <RepoCard key={index} fullName={repo.full_name} githubLogoUri={githubLogoUri} />;
                      })
                    : null}
                  {}
                </div>
              </div>
              <div className="mt-5 sm:mt-4 sm:flex sm:flex-row-reverse">
                <button
                  type="button"
                  className="w-full inline-flex justify-center rounded-md border border-transparent shadow-sm px-4 py-2 bg-dolist-darkblue dark:bg-dolist-green text-base font-std font-bold text-white focus:outline-none focus:ring-2 focus:ring-offset-2 sm:ml-3 sm:w-auto sm:text-sm"
                  onClick={async (e) => {
                    await addRepoSubmitHandler(e);
                    setOpen(false);
                  }}
                >
                  Confirm
                </button>
              </div>
            </div>
          </Transition.Child>
        </div>
      </div>
    </Transition.Root>
  );
};

export default AddRepoModal;
