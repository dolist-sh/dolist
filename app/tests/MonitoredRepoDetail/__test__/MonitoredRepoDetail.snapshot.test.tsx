import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { MonitoredRepoDetail } from '../../../src/components';
import { MonitoredRepo } from '../../../src/types';

describe('Snapshot testing of MonitoredRepoDetail component', () => {
  const testData = {
    theme: 'dark',
    repo: {
      createdAt: 1664701600,
      defaultBranch: 'main',
      fullName: 'whathecker/assignment-restapi',
      id: 'f5e6ce58-636f-4cc9-b150-46754e70cd46',
      language: 'Python',
      lastCommit: 'bfae0103c24a05e8d77601bad8bf647b2500663f',
      lastUpdated: 1666007886,
      name: 'assignment-restapi',
      parsedComments: [],
      provider: 'github',
      status: 'active',
      userId: '68202eee-a8b5-4348-a146-55cef2c2bb52',
      visibility: 'public',
    } as MonitoredRepo,
    githubLogoUri: 'http://awesomelogo',
  };

  test('renders correctly', () => {
    const component = renderer.create(
      <MonitoredRepoDetail theme={testData.theme} repo={testData.repo} githubLogoUri={testData.githubLogoUri} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
