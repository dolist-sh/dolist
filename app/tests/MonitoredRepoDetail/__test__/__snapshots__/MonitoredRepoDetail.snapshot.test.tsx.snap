// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing of MonitoredRepoDetail component renders correctly 1`] = `
Array [
  <div
    className="flex flex-row w-[80%] m-auto mt-7 pl-6 pr-6 justify-evenly"
  >
    <div
      className="flex flex-row w-1/2"
    >
      <img
        className="w-9 h-9 self-center mr-3"
        src="http://awesomelogo"
      />
      <div
        className="flex flex-col"
      >
        <h4
          className="font-std font-bold text-[14px] text-black dark:text-dolist-cream"
        >
          whathecker/assignment-restapi
        </h4>
        <p
          className="font-std font-bold text-[10px] text-dolist-gray dark:text-dolist-cream"
        >
          branch: 
          <span
            className="font-std font-bold underline text-dolist-darkblue dark:text-white"
          >
            main
          </span>
        </p>
      </div>
    </div>
    <div
      className="flex flex-row-reverse w-1/2"
    >
      <p
        className="text-[11px] font-std font-bold text-right text-dolist-gray dark:text-dolist-cream"
      >
        from last commit: 
        <span
          className="font-std font-bold underline text-dolist-darkblue dark:text-white"
        >
          bfae010
        </span>
      </p>
    </div>
  </div>,
  <div
    className="flex flex-col w-[76%] m-auto mt-9 border-b border-dolist-lightgray dark:border-dolist-cream"
  >
    <div
      className="flex flex-row pl-2 pb-2"
    >
      <p
        className="cursor-pointer text-xs font-std font-bold text-dolist-gray dark:text-dolist-cream pr-6"
      >
        Total 0
      </p>
      <p
        className="cursor-pointer text-xs font-std hover:font-bold text-dolist-gray dark:text-dolist-cream pr-6"
      >
        New 0
      </p>
      <p
        className="cursor-pointer text-xs font-std hover:font-bold text-dolist-gray dark:text-dolist-cream pr-6"
      >
        Resolved 0
      </p>
      <p
        className="cursor-pointer text-xs font-std hover:font-bold text-dolist-gray dark:text-dolist-cream pr-6"
      >
        Old 0
      </p>
    </div>
  </div>,
  <div
    className="flex flex-col w-[76%] m-auto mt-9"
  />,
]
`;
